fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -2514,39 +1845,45 @@ public static OnlineIndexer forRecordStoreAndIndex(@Nonnull FDBRecordStore recor
         /**
          * Only build if the index is disabled.
          */
-        BUILD_IF_DISABLED(false),
+        BUILD_IF_DISABLED(false, OnlineIndexerCommon.IndexStatePrecondition.BUILD_IF_DISABLED),
```

Comment:
I understand in the design you don't want to make `OnlineIndexerCommon` to depend on `OnlineIndexer`. But here I think it is fine to make `IndexStatePrecondition` **static** and import it into `OnlineIndexerCommon`.